File: Data Structures & Algorithms/find-pivot-index/submission-1.java
class Solution {
    public int pivotIndex(int[] nums) {
        int totalSum = Arrays.stream(nums).sum();
        
        int currSum = 0;
        for(int i = 0; i < nums.length; i++) {
            int leftSideSum = currSum;
            currSum += nums[i];
            int rightSideSum = totalSum - currSum;
            if (leftSideSum == rightSideSum) {
                return i;
            }
        }

        return -1;
    }
}

File: Data Structures & Algorithms/continuous-subarray-sum/submission-1.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        if len(nums) < 2:
            return False
        if len(nums) == 2:
            return sum(nums) % k == 0
        
        remainder_idx = {}
        curr_sum = 0
        for i, num in enumerate(nums):
            curr_sum = (curr_sum + num) % k
            if curr_sum == 0 and i >= 1:
                return True
            if curr_sum in remainder_idx and i - remainder_idx[curr_sum] > 1:
                return True
            elif curr_sum not in remainder_idx:
                remainder_idx[curr_sum] = i
        return False